PyRDF/backend/Spark.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import print_function
+from .Dist import Dist
+from pyspark import SparkConf, SparkContext
+
+class Spark(Dist):
+    """
+    Backend that executes the computational graph
+    using using `Spark` framework for distributed
+    execution.
+
+    """
+    def __init__(self, config = {}):
+        """
+        Creates an instance of the Spark
+        backend class.
+
+        Parameters
+        ----------
+        config : dict (optional)
+            The config options for Spark backend. The default
+            value is an empty Python dictionary `{}`. Config
+            specifically reads three options, 'SparkContext'
+            or 'SparkConf' and 'npartitions'. If 'SparkContext'
+            is not given, 'SparkContext' uses 'SparkConf' option,
+            which by default is 'None' and 'npartitions' is
+            '2' by default.
+
+        """
+        super(Spark, self).__init__(config)
+
+        try:
+            # Remove the value of 'npartitions' from config dict
+            self.npartitions = config.pop('npartitions')
+        except KeyError:
+            # Set the default value to 2 if 'npartitions'
+            # doesn't exist
+            self.npartitions = 2
+
+        sparkConf = SparkConf().setAll(config.items())
+        self.sparkContext = SparkContext.getOrCreate(sparkConf)
```

Comment:
If a SparkContext exists (like the case of SWAN), these configs are ignored. I guess there's no problem, as the user was the one that created the config. Do you imagine any problem with this?
Should we check if there's already a sc and give a warning or it's not really necessary?

Reply:
@diocas , yes. That'd be a useful message. It'd be great to show a warning. Let's wait and see what @etejedor has to say about this one

Reply:
Yes, I agree we should show a warning if a `SparkContext` already exists and the user is trying to define new configurations.